jujugui/static/gui/src/app/components/inspector-config/inspector-config.js

```diff
@@ -21,8 +21,81 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 YUI.add('inspector-config', function() {
 
   juju.components.Configuration = React.createClass({
+    /**
+      Set the intial state.
+
+      @method getInitialState
+      @returns {String} The current state.
+    */
+    getInitialState: function() {
+      return {
+        // Have to clone the config so we don't update it via reference.
+        serviceConfig: JSON.parse(JSON.stringify(
```

Comment:
Because this stringify/parse dance is done here and in `componentWillReceiveProps` I think it should be copied out into a 'clone' fn just so that we don't accidentally miss updating one or the other in the future.